meson.build

```diff
@@ -163,9 +167,21 @@ if get_option('b_sanitize') != 'none'
     ]
 endif
 
+if get_option('buildtype') == 'release'
+    default_cpp_flags += [
+        '-z relro',
+        '-z noexecstack',
+    ]
+    default_c_flags += [
+        '-z relro',
+        '-z noexecstack',
+    ]
```

Comment:
Heh, things you learn: the compiler accepts `-z`. I knew this only as a linker option and that you'd need to spell them `-Wl,-z,relro -Wl,-z,noexecstack`. But since they are linker options (and as is `-pie`), you may need to use the `add_project_link_arguments()` Meson function.

Also, this shouldn't be for release builds only, so go ahead and enable for everything.